### PR TITLE
Use RawFile in EnsimeConfig

### LINKED ISSUE #1791: config should use RawFile, not File
Pretty simple refactor, will touch lots of code.

It would be good to take the opportunity to push for more nio / `Path` usage instead of legacy IO as per #1437

### PR DESCRIPTION
**Pull request by zaynetro**
Closes #1791 

Do we want to use `RawFile` in `EnsimeProject` too?

### CODE REVIEW

File: server/src/main/scala/org/ensime/server/Server.scala [hunk 1]
```diff
@@ -80,7 +80,7 @@ class ServerActor(
         log.info(s"ENSIME HTTP on ${ch.localAddress()}")
         try {
           val port = ch.localAddress().asInstanceOf[InetSocketAddress].getPort()
-          PortUtil.writePort(config.cacheDir, port, "http")
+          PortUtil.writePort(config.cacheDir.file.toFile, port, "http")
```
Comment (fommil) [inline server/src/main/scala/org/ensime/server/Server.scala]: it would be good to update this method to take `Path` instead of `File`

**Comment by fommil** [comment]
yes definitely, please use `RawFile` instead of `File` everywhere you can.

File: core/src/it/scala/org/ensime/core/RichPresentationCompilerSpec.scala [hunk 2]
```diff
@@ -598,7 +598,7 @@ trait RichPresentationCompilerTestUtils {
   }
 
   def srcFile(proj: EnsimeConfig, name: String, content: String, write: Boolean = false, encoding: String = "UTF-8"): BatchSourceFile = {
-    val src = proj.projects.head.sources.head / name
+    val src = proj.projects.head.sources.head.file.resolve(name).toFile
```
Comment (fommil) [inline core/src/it/scala/org/ensime/core/RichPresentationCompilerSpec.scala]: if you import our `org.ensime.util.path._` helpers, we should have `/` for `Path`

Comment (zaynetro) [inline core/src/it/scala/org/ensime/core/RichPresentationCompilerSpec.scala]: Thanks! There is a plenty of useful methods there.

Comment (fommil) [inline core/src/it/scala/org/ensime/core/RichPresentationCompilerSpec.scala]: you also get some methods on the `Rawfile` by importing `org.ensime.util.ensimefile._`

**Comment by zaynetro** [comment]
I will look into `org.ensime.util.ensimefile._` next

**Comment by zaynetro** [comment]
Updated to reuse `ensimefile` util methods

**Comment by zaynetro** [comment]
It all starts with a broken test

**Comment by zaynetro** [comment]
I will take a look at this again during next week.

**Comment by zaynetro** [comment]
OK. Seems like I finally figured everything that went wrong.

**Comment by zaynetro** [comment]
@fommil This one is done

**Comment by fommil** [comment]
Sweet! I'll take a look in a few days, ping me if I don't. I'm on holiday now.

**Pull request merged**